Posts/2021/02/ScriptingGuy.0002.GetDateOfYesterday.md

```diff
@@ -0,0 +1,185 @@
+Then we can look at how to calculate yesterday and use that in your scripts.
+
+## Dates in PowerShell
+
+Let's start by looking at how you can deal with dates and times.
+As you probably know, PowerShell contains the `Get-Date` cmdlet.
+This cmdlet returns a .NET **System.DateTime** object.
+
+Using the `Get-Date` cmdlet, you can get any date and time, and either display it or store it in a variable. 
+To get today's date. you could do this:
+
+```powershell
+# Get the current date
+Get-Date
+# Store the date in a variable then view it
+$Now = Get-Date
+$Now
+```
+
+The output looks like this:
+
+```powershell-console
+PS C:\> # Get the current date
+PS C:\> Get-Date
+08 January 2021 11:24:46
+
+# Store the date in a variable
+$Now = Get-Date
+$Now
+08 January 2021 11:24:47
+```
+
+As mentioned, the `Get-Date` cmdlet returns an object whose type is **System.DateTime**.
+This .NET structure provides a rich set of properties and methods to help you manipulate the date/time object.
+See [System.DateTime documentation](https://docs.microsoft.com/dotnet/api/system.datetime) for more details on this structure.
+A date and time object contains both a date and a time.
+This means you can create an object with just a date or just a time, or both, which gives you huge flexibility in handling dates and times.
+
+If you run `Get-Date` and specify no parameters, the cmdlet returns the current date and time.
+THere are several parameters you can specify that allow you to create an object for a particular date, like this:
+
+```powershell
+# Using the -Date Parameter and a date string
+Get-Date -Date '1 August 1942'
+# Using parameters to Get-Date to specify specific month, day, and year independently.
+Get-Date -Month 8 -Day 1 -Year 1942
+```
+
+The output of these two commands is the same:
+
+```powershell-console
+PS C:\> Get-Date -Date '1 August 1942'
+01 August 1942 00:00:00
+
+PS C:\> > Get-Date -Month 8 -Day 1 -Year 1942 -Hour 0 -Minute 0 -Second 0
+01 August 1942 00:00:00
+```
+
+You can see the other features of `Get-Date` to help get the date in the exact format you need, see the [`Get-Date` help information](https://docs.microsoft.com/powershell/module/microsoft.powershell.utility/get-date?view=powershell-7.1).
+
+## Obtaining Yesterday's Date
+
+You can use `Get-Date` to return a specific date/time.
+So how do you get yesterday's date - or the date a date of yesterday, or last month or last year?
+The trick here is to use the object returned from `Get-Date`.
+The object has a type of `System.DateTime` which contains a number of methods allowing you to add increments of time - a month, a day, etc to the object.
+
+To get yesterday's date (or tomorrow's) you create a date and time object for today using `Get-Date` with no parameters.
+Then you use the ``AddDays`` method to add/subtract some number of days, like this:
+
+```powershell
+# Get today's sate
+$Today = Get-Date
+$Yesterday = $Today.AddDays(-1)
+$Yesterday
+# Or more simply
```

Comment:
Can you add a new line between examples for better readability line 83 and 86?

Reply:
added